src/core/meshcop/secure_transport.hpp

```diff
@@ -104,9 +104,10 @@ class SecureTransport : public InstanceLocator
      *
      * @param[in]  aContext    A pointer to application-specific context.
      * @param[in]  aConnected  TRUE if a connection was established, FALSE otherwise.
+     * @param[in]  aWithError  TRUE if a connection was torn down due to an error.
      *
      */
-    typedef void (*ConnectedHandler)(void *aContext, bool aConnected);
+    typedef void (*ConnectedHandler)(void *aContext, bool aConnected, bool aWithError);
```

Comment:
Instead of adding a new boolean `aWithError` suggest we define an `enum` and use this in the callback

For example:
```c++
    enum ConnectEvent : uint8_t
    {
        kConnected,                    ///< Connection established.
        kDisconnectedPeerClosed,       ///< Disconnected by peer.
        kDisconnectedLocalClosed,      ///< Disconnected locally.
        kDisconnectedMaxConnAttempts,  ///< Disconnected due to reaching the maximum number of failed connection attempts.
        // ... other events
    };

    /**
     * Pointer is called to report a connection event, e.g., connection is established or torn down.
     *
     * @param[in]  aContext    A pointer to application-specific context.
     * @param[in]  aEvent      The connection event.
     *
     */
    typedef void (*ConnectedHandler)(void *aContext, bool ConnectEvent aEvent)
```

Using `enum` will be more flexible (can add more cases in future) and improves readability, e.g., invoking callbacks `mConnectedCallback.InvokeIfSet(true, false) -> mConnectedCallback.InvokeIfSet(kConnected)`.

Reply:
Updated to enum